sympy/physics/control/lti.py

```diff
@@ -0,0 +1,121 @@
+from sympy import (
+    Basic, Matrix, ShapeError, Mul,
+    ImmutableMatrix, MutableMatrix, MutableDenseMatrix,
+)
+from sympy.polys.polytools import degree
+
+__all__ = ['TransferFunctionMatrix', 'TransferFunction']
+
+_matrixTypes = (
+    Matrix, ImmutableMatrix, MutableMatrix, MutableDenseMatrix)
+
+
+class TransferFunctionMatrix(Basic):
+    def __new__(cls, args):
+        # constructor is from a given transfer function.
+        if  isinstance(args, (Matrix, ImmutableMatrix, MutableMatrix)):
+            G = args
+            obj = Basic.__new__(cls, G)
```

Comment:
Matrix and MutableMatrix are the same so you don't need to test them separately. Also Matrix sympifies to ImmutableMatrix so it's better to use `_sympify` and then check that the result is an `ImmutableMatrix`. Mutable matrix should never be in the args of `Basic`.

Reply:
Thanks! I'll eventually remove `TransferFunctionMatrix` from this PR, and it will be dealt with in the following PRs. Will address these comment there.